### PR TITLE
Implemented elementwise softmax cross entropy loss

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented ElementwiseSoftmaxCrossEntropy which calculates softmax cross entropy for each element. SoftmaxCrossEntropy can be replaced with this new function and sum and div.

### CODE REVIEW

**Comment by unnonouno** [comment]
And `class_weight` can be written as simple product. Is it required for this function?

**Comment by MannyKayy** [comment]
Is it possible to add native support for masking over the mini-batch elements, in the cases where we may want to ignore them?

**Comment by delta2323** [comment]
@MannyKayy `SoftmaxCrossEntropy` function supports `ignore_label` option for this purpose. If a sample has a label with this value, this sample is ignored in summing up loss values.

**Comment by delta2323** [comment]
@unnonouno Could you write a document about `reduce` option?

**Comment by unnonouno** [comment]
OK, i updated it.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 1]
```diff
@@ -243,9 +243,19 @@ class weight ``class_weight[i]`` is actually multiplied to
             value.
         ignore_label (int): Label value you want to ignore. Its default value
             is ``-1``. See description of the argument `t`.
+        reduce (str): Variable holding a ``str`` which
```
Comment (delta2323) [inline chainer/functions/loss/softmax_cross_entropy.py]: `reduce` option is not a `Variable`. So "A string that determines ..." is better.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 2]
```diff
@@ -243,9 +243,19 @@ class weight ``class_weight[i]`` is actually multiplied to
...
+            determines whether to reduce the shape of the input.
```
Comment (delta2323) [inline chainer/functions/loss/softmax_cross_entropy.py]: As the input shape is not the target of reduction, it is better "... to reduce loss values".

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 3]
```diff
@@ -243,9 +243,19 @@ class weight ``class_weight[i]`` is actually multiplied to
...
+            If it is ``'mean'``, it computes the sum of cross entropy
```
Comment (delta2323) [inline chainer/functions/loss/softmax_cross_entropy.py]: Suggestion: ... sum of the individual cross entropy.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 4]
```diff
@@ -243,9 +243,19 @@ class weight ``class_weight[i]`` is actually multiplied to
...
+            and normalize it according to ``normalize`` option.
+            If it is ``'no'``, this function computes cross entropy for each
+            instance and does not normalize it (``normalize`` option is
+            ignored). In this case, the loss value of the ignored instance,
+            which has ``ignore_label`` as its target value, is set to ``0``.
 
     Returns:
         Variable: A variable holding a scalar array of the cross entropy loss.
+        If ``reduce`` is ``'mean'``, it is a scalar array.
+        If ``reduce`` is ``'no'``, the shape is same as ``x``.
```
Comment (delta2323) [inline chainer/functions/loss/softmax_cross_entropy.py]: ... same as that of ``x``.

**Comment by delta2323** [comment]
Check the function certainly throws error when invalid options are given to `reduce` option.

**Comment by delta2323** [comment]
LGTM
